scripts/check_runtime

```diff
@@ -0,0 +1,294 @@
+
+      next unless matches
+
+      tag = matches[:tag]
+      next unless tag
+
+      case tag
+      when "method_missing", "shim"
+        unless matches[:desc]
+          $stderr.puts(" * Annotation `@#{tag}` requires a description (#{comment.loc})")
+        end
+      end
+
+      annotations << tag
+    end
+
+    annotations
+  end
+
+  def scope_name(scope)
+    scope_name = case scope
+    when RBI::Class, RBI::Module
+      scope_name = scope.fully_qualified_name
+    when RBI::SingletonClass
+      scope_name = scope.parent_scope&.fully_qualified_name
+    end
+
+    scope_name || "Object"
+  end
+
+  class Context
+    TEST_NAME = "test.rb"
+
+    def initialize(gem_name)
+      @gem_name = gem_name
+      @gemfile = String.new
+      @requires = String.new
+      @body = String.new
+      @workdir = Dir.mktmpdir
+    end
+
+    def run!
+      write_gemfile!
+      write_test!
+
+      out, status = bundle_install!
+      unless status.success?
+        $stderr.puts(" * Can't install gem `#{@gem_name}` (#{out.gsub(/\n/, "")})")
+        return false
+      end
+
+      out, status = run_test!
+      unless status.success?
+        $stderr.puts(out)
+        return false
+      end
+
+      true
+    ensure
+      FileUtils.rm_rf(@workdir)
+    end
+
+    def add_gem_dependency(name)
+      @gemfile << <<~RB
+        gem "#{name}"
+      RB
+    end
+
+    def add_require(name)
+      @requires << <<~RB
+        begin
+          require "#{name}"
+        rescue LoadError => e
+          $stderr.puts(" * Can't require `#{name}`")
+          $success = false
+        end
+      RB
+    end
+
+    def add_constant(const_name, loc)
+      @body << <<~RB
+        __rbi_repo_get_const("#{const_name}", "#{loc.to_s}")
+      RB
+    end
+
+    def add_method(recv_name, method_name, loc, allow_missing:, singleton: false)
+      @body << <<~RB
+        __rbi_repo_get_method(
+          "#{recv_name}",
+          "#{method_name}",
+          "#{loc.to_s}",
+          singleton: #{singleton},
+          allow_missing: #{allow_missing}
+        )
+      RB
+    end
+
+    private
+
+    def write_gemfile!
+      File.write("#{@workdir}/Gemfile", <<~GEMFILE)
+        source "https://rubygems.org"
+        #{@gemfile}
+        gem "#{@gem_name}"
+      GEMFILE
+    end
+
+    def write_test!
+      File.write("#{@workdir}/#{TEST_NAME}", ruby_string)
+    end
+
+    def bundle_install!
+      Bundler.with_unbundled_env do
+        Open3.capture2e("bundle config set --local path 'vendor/bundle'", chdir: @workdir)
+        Open3.capture2e("bundle install --quiet", chdir: @workdir)
+      end
+    end
+
+    def run_test!
+      Bundler.with_unbundled_env do
+        Open3.capture2e("bundle exec ruby #{TEST_NAME}", chdir: @workdir)
+      end
+    end
+
+    def ruby_string
+      <<~RB
+        $success = true
+
+        def __rbi_repo_get_const(const_name, rbi_loc)
+          Kernel.const_get("\#{const_name}")
```

Comment:
Similarly here we could use [const_defined?](https://ruby-doc.org/core-3.1.2/Module.html#const_defined-3F-method) to not rely on exceptions.